docs/analyzer-rules/AZFW0012.md

```diff
@@ -0,0 +1,28 @@
+# AZFW0012: Invalid Retry Options
+
+| | Value |
+|-|-|
+| **Rule ID** |AZFW0012|
+| **Category** |[AzureFunctionsSyntax]|
+| **Severity** |Error|
+
+## Cause
+
+This rule is triggered when a retry attribute is incorrectly used on an Azure Function.
+
+## Rule description
+
+This rule is triggered in two scenarios:
+
+- A retry strategy is not recognized (only Fixed Delay and Exponential Backoff are supported)
+- A retry attribute is used on a trigger type that does not support function-level retry.
+
+A list of triggers that support function-level retry can be found [here](https://learn.microsoft.com/en-us/azure/azure-functions/functions-bindings-error-pages?tabs=fixed-delay%2Cin-process&pivots=programming-language-csharp#retries).
+
+## How to fix violations
```

Comment:
nit: Should we also provide sample for correct usage here / point to valid samples in docs?